src/net_peer.rs

```diff
@@ -192,25 +192,39 @@ impl Drop for MyTcpStream {
     }
 }
 
-pub fn tcp_connect_peer(addr: &SocketAddr) -> BoxedNewPeerFuture {
-    Box::new(
+pub fn resolve_addr(
+    addr_str: &str,
+) -> impl Future<Item = SocketAddr, Error = Box<dyn std::error::Error + 'static>> {
+    // TODO: resolve asynchronously
```

Comment:
Is it really needed to be resolved at connecting time, not just once at argument parsing time?

Reply:
I think it is: when running websocat as a daemon, it shouldn’t need to be restarted just because a remote DNS name changed. Additionally, one might want to point websocat at a service using DNS load balancing.

Reply:
Tardiness in DNS reply may not only delay establishing a new connection, but also suspend all unrelated neighbour connections - websocat is mostly single-threaded.

Reply:
Yeah, until we get async DNS, that’s a tradeoff. But this design allows the user to make that tradeoff. They can always choose to provide an IP address, or to install a local caching resolver (which will do the job much better than we would, respecting TTLs and so on).